### ISSUE TITLE
Makefile: test: find -executable not a valid BSD find argument

### ISSUE THREAD

**Issue by rickard-von-essen** [hashicorp/packer#7243]
Unfortunately BSD find doesn't have `-executable` so this check is in practice skipped on macOS and other BSD's. Introduced in 03620d15c2f005783ccba7f5ab78aad1baf49c6c

```
$ make test
==> Checking that code complies with gofmt requirements...
Check passed.
find: -executable: unknown primary or operator
find: -executable: unknown primary or operator
==> Checking that only certain files are executable...
```

**Comment by azr** [comment]
Yes, the commit should have been clearer here. I went for the `-executable` command as it works in [circle-ci](https://circleci.com/gh/hashicorp/packer/358?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) and appveyor [1] and I would rather have the ci-tools have the last word. At HashiCorp, we are making circle-ci the main ci tool and [the `-perm` parameter doesn't work in circle-ci](https://circleci.com/gh/hashicorp/packer/362?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

That said, I would also rather have it work everywhere.
One thing we can do is check wether the system is bsd/darwin and switch the option in that case.

1: that said appveyor doesn't run make ci ( yet ? )

**Comment by rickard-von-essen** [comment]
> One thing we can do is check wether the system is bsd/darwin and switch the option in that case.

That is probably the best solution.

**Comment by kp2099** [comment]
Closing this issue as the Makefile has been changed and does not use -executable anymore.
